rsmtool/utils/prmse.py

```diff
@@ -0,0 +1,328 @@
+"""
+Utility classes and functions related to computing test
+theory based evaluations.
+
+The derivations and formulas were provided by Matt Johnson.
+
+:author: Anastassia Loukina (aloukina@ets.org)
+:author: Nitin Madnani (nmadnani@ets.org)
+
+:organization: ETS
+"""
+
+import pandas as pd
+import numpy as np
+
+
+def get_n_human_scores(human_scores):
+    """
+    Compute number of human scores for each response
+
+    Parameters
+    ----------
+    human_scores : array-like of shape (n_samples, n_ratings)
+        Human ratings for each response
+
+    Returns
+    -------
+    n_scores : array-like of shape (n-samples)
+        Total number of not None human scores
+    """
+    n_scores = (~np.isnan(human_scores)).sum(axis=1)
+    return n_scores
+
+
+def variance_of_errors(human_scores):
+    '''Compute variance of human errors.
+
+    Parameters
+    ----------
+    human_scores : array-like of shape (n_samples, n_ratings)
+        Human ratings for each response
+
+    Returns
+    -------
+    variance_of_errors : float
+        Estimated variance of human errors
+    '''
+
+    # if the user passsed pandas Dataframe,
+    # we convert it to an array since
+    # in these computations having column names
+    # interferes with the computation
+
+    if isinstance(human_scores, pd.DataFrame):
+        human_scores = human_scores.to_numpy()
+
+    # we only use responses with more than 1 score
+
+    n_scores = get_n_human_scores(human_scores)
+
+    multiple_mask = n_scores > 1
+
+    # raise an error if we don't have any such responses
+    if multiple_mask.sum() == 0:
+        raise ValueError("Variance of human errors "
+                         "necessary for true score "
+                         "evaluations requires "
+                         "at least a subset of responses "
+                         "to be scored by 2 or more "
+                         "raters.")
+
+    multiple_scores = human_scores[multiple_mask]
+
+    n_scores = n_scores[multiple_mask]
+
+    variances = np.nanvar(multiple_scores, ddof=1, axis=1)
+
+    variance_of_errors = np.average(variances, weights=n_scores - 1)
+
+    return variance_of_errors
```

Comment:
```suggestion
   # compute the pooled-variance estimator
   
   # get the responses that have more then one human rating
    multiple_scores = human_scores[multiple_mask]
    
    # how many ratings does each such response have
    n_scores_for_multiple = n_scores[multiple_mask]
    
    # compute the pooled variance with a single degree of freedom
    variances = np.nanvar(multiple_scores, ddof=1, axis=1)
    
    # compute the weighted average of these variances
    variance_of_errors = np.average(variances, weights=n_scores_for_multiple - 1)
    return variance_of_errors
```

I think it's confusing to use the same variable name again plus it'd be nice to have some comments,